python/pyspark/ml/tuning.py

```diff
@@ -229,6 +228,24 @@ def fit(self, dataset, params={}):
         bestModel = est.fit(dataset, epm[bestIndex])
         return CrossValidatorModel(bestModel)
 
+    def copy(self, extra={}):
+        """
+        Creates a copy of this instance with a randomly generated uid
+        and some extra params. This copies the underlying estimator,
+        evaluator, and estimatorParamMap, creates a deep copy of the
+        embedded paramMap, and copies the embedded and extra parameters
+        over.
+        :param extra: Extra parameters to copy to the new instance
+        :return: Copy of this instance
+        """
+        newCV = Params.copy(self, extra)
```

Comment:
I realized that, while I believe this works, it will make a deep copy of the Estimator and Evaluator, which is a little inefficient.  Not that important